packages/react-i18n/src/babel-plugin/README.md

```diff
@@ -0,0 +1,266 @@
+# `@shopify/react-i18n/babel`
+
+This package includes a plugin for Babel that auto-fills `useI18n`'s or `withI18n`'s arguments from an adjacent translations folder.
+
+## Installation
+
+```bash
+$ yarn add @shopify/react-i18n
+```
+
+## Usage
+
+The Babel plugin is exported from the `@shopify/react-i18n/babel` entrypoint:
+
+```js
+// babel.config.js
+{
+  plugins: [
+    ['@shopify/react-i18n/babel'],
+  ],
+}
+```
+
+This plugin will look for an adjacent _`translations`_ folder containing, at minimum, an `en.json` file (the default locale). It will then iterate over each reference to the `useI18n` hook or `withI18n` decorator and, if the reference is a call expression with no arguments, and inject the appropriate arguments.
+
+### Default Transform
+
+Given folder structure
+
+```
+├── MyComponent.tsx
+└── translations
+   └── en.json
+   └── fr.json
+   └── jp.json
+
+```
+
+File content
+
+```js
+// Within translations/en.js
+
+{
+  "heading": "All Films"
+}
+
+// Within translations/fr.js
+
+{
+  "heading": "fr_All Films"
+}
+
+// Within translations/jp.js
+
+{
+  "heading": "jp_All Films"
+}
+```
+
+```js
+// Within MyComponent.tsx:
+
+useI18n();
+
+// Becomes:
+
+import _en from './translations/en.json';
+
+useI18n({
+  id: 'MyComponent_<hash>',
+  fallback: _en,
+  async translations(locale) {
+    if (['fr', 'jp'].indexOf(locale) < 0) {
+      return;
+    }
+
+    return (async () => {
+      const dictionary = await import(
+        /* webpackChunkName: "MyComponent_<hash>-i18n", webpackMode: "lazy-once" */ `./translations/${locale}.json`
+      );
+      return dictionary && dictionary.default;
+    })();
+  },
+});
+```
+
+## Options
+
+### `mode`
+
+Type: `undefined`, `from-generated-index`, `from-dictionary-index`
+
+Default: `undefined`
+
+#### `from-generated-index`
+
+_*Use this mode to integration with babel-loader.*_
+
+Because `babel-loader`'s cache is based on a component's source content hash, newly added translation files will not invalidate the component's Babel cache. To combat this, run the `generateTranslationIndexes` function before building, and configure the plugin to use its `from-generated-index` mode.
+
+The generator will look for any `translations` folders and generate an array of local ids in `translations/index.js` based on the `{locale}.json` files found. We recommend that you add `**/translations/index.js` to `.gitignore` to make sure the generated files are not checked-in.
+
+##### Usage
+
+```js
+// webpack.config.js
+module.exports = {
+  resolve: {
+    extensions: ['.js', '.jsx'],
+  },
+  module: {
+    rules: [
+      {
+        test: /\.jsx?$/,
+        use: [
+          {
+            loader: 'babel-loader',
+            options: {
+              plugins: [
+                '@babel/plugin-syntax-dynamic-import',
```

Comment:
In the first config demo, we show babel.config.js style config here we switch to webpack, and in the final example we claim "webpack.config.js" but the code looks the same as the first.

Can we keep a consistent style. I like the one used in the first config. So this block would be 👍 

```js
plugins: [
  ['@shopify/react-i18n/babel', {mode: 'from-generated-index'}],
],
```

Reply:
I am actually going to change it all to `webpack.config.js` instead since webpack is actually a dependency on the default transform